### ISSUE TITLE
Metric measurements

### ISSUE THREAD

**Issue by spaceconquistador** [crnormand/gurps#188]
It would be great if there was a way of changing the all the displayed measurements on the character sheet to metric forms of measurements.

**Comment by crnormand** [comment]
Thank you for the suggestion.  Obviously this would depend on the data coming from the character creation tools (GCA/GCS).   

Encumbrance "Max Load" is a integer from both.

"Living and Moving things" values are integers from GCA but strings (that include "lb") from GCS.   ex:  "45 lb"

GCS inventory is mixed:

        <name type="string">Copper Coin</name>
        <isidentified type="number">1</isidentified>
        <count type="number">25</count>
        <cost type="number">1</cost>
        <costsum type="string">25</costsum>
        <weight type="number">0.02</weight>
        <weightsum type="number">0.5 lb</weightsum>

"weightsum" is the only thing that includes units.

Whereas GCA's inventory doesn't include the "sums" data at all:

<name type="string">Traveler's Rations</name>
<count type="number">10</count>
<cost type="string">2</cost>
<weight type="number">0.5</weight>

GCS provides height and weight as string that include the measurements.

And finally, the measuring tool shows "yds" per hex.   This is defined in the system.json, so I don't know how (or if) we can make this an option:

  "gridDistance": 1,
  "gridUnits": "yd",

So yeah, I think we could do this... assuming GCA and/or GCS can output metric.

**Comment by gtludwig** [comment]
What if importer only considers the numeric part of an alphanumeric string from a "type=number" node, for instance in the <weightsum /> node. This could be achieved by a regex matcher. 
Another regex could match the unit (lb, kg, etc.) and "save" that entry for that field. 
So, once imported, you would have a map like 
itemsMap: {
  item: {
    key: "some-hash",
    value: {
      name: "Copper coin",
      quantity: 25,
      weight: {
        weight: 0.02,
        unit: "lb"
      },
     weightsum: {
       weightsum: 0.5,
       unit: "lb"
     }
    }
  }, 
}
Afterwards, we could look into a converter.

**Comment by crnormand** [comment]
Certainly, that is exactly what I was thinking.   We can parse the data, the real issue is what data will we be parsing?   GCS has a bunch of options for units for weights:   short tons (tn), long tons (lt), metric tons (t), kilograms (kg) and grams (g) and pounds (lb).   Unfortunately, it does not currently have an export key to tell us which unit it is using.

And, as far as I can tell, GCA does not have any option to use metric values (although, I am nowhere near an expert on this...so I have asked about that in the Discord channel).

My thought is that we could offer 3 settings:
Import in Imperial (lbs), display Imperial (lbs)
Import in Imperial (lbs), display Kilograms (kg) (and do the conversion ourselves)
Import in Kilograms (kg), display Kilograms (kg)

**Comment by crnormand** [comment]
I checked, and GCA can do metric, if you use the metric data files.   Since I am not certain how many people will do so, I think we are left with the 3 options above (and let the GM decide).

**Comment by crnormand** [comment]
the mod bucket would need to convert miles at longer ranges into km

**Comment by cefasheli** [comment]
You can change this information /m/km/Kg in Edit\Global Preferences in GCS

**Comment by crnormand** [comment]
Yes, but the problem is that in Foundry, we still will show "LBS", even if GCS is using KG.   GCS does not tell us which "system" it is using (Imperial or Metric) so we don't know which one to display (and currently, everything is hardcoded to be Imperial)

**Comment by Pimpaum** [comment]
Hey guys, I used metric in GCS, imported the character sheet into Foundry and got some mixed results. Note that the equipment weight was converted back into imperial but the encumbrance table remained as metric. Besides the problem of dealing with two diferent systems, the encumbrance level was also miscalculated: the total equipment weight is 14lbs (~7kg) but my encumbrance level is marked as "light" for it considers I am carrying more than 12kg (~24lbs).

My guess is that it is using comparing 14lbs with the 12kg treshold and ignoring the conversion. So with 14 being bigger than 12, my charecter is on light encumbrance


![image](https://user-images.githubusercontent.com/80008673/109876841-056a9b80-7c51-11eb-81be-8595a0fbd36b.png)
![image](https://user-images.githubusercontent.com/80008673/109877593-f7694a80-7c51-11eb-816b-7a64260fb5ac.png)

**Comment by crnormand** [comment]
The problem is that for some information (like encumbrance) GCS includes the "kg"), but for others, it does not... and we don't convert it.   So we have some stuff hardcoded to "LBS" in our character sheets and code, and some data is just strings that we get from the character creation tool.

I am sorry, but for the moment, it would be best to remain in Imperial (at least then, all of the values match).

We plan to offer a system setting that will allow us to:

Import in Imperial (lbs), display Imperial (lbs) -- the current default
Import in Imperial (lbs), display Kilograms (kg) (and do the conversion ourselves) - For GCA users who cannot easily do metric
Import in Kilograms (kg), display Kilograms (kg)  -- allow GCS to be in metric and just use its values (and labels)

But until then... you will get the mashup as above ;-)

**Comment by Pimpaum** [comment]
All right, thank you!
I´ll do as recommended. :-)
